### PR TITLE
ROX-12015: Configure OSCI branch for stackrox upstream build

### PR DESCRIPTION
**Pull request by 0x656b694d**
## Description

A new step in "Start Release" automation.

## Checklist
- [ ] ~Investigated and inspected CI test results~
- [ ] ~Unit test and regression tests added~
- [ ] ~Evaluated and added CHANGELOG entry if required~
- [ ] ~Determined and documented upgrade steps~
- [ ] ~Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

## Testing Performed

Locally and in `test-gh-actions` repository:
https://github.com/stackrox/test-gh-actions/actions/runs/2912915223

### CODE REVIEW

File: .github/workflows/start-release.yml [hunk 1]
```diff
@@ -109,6 +109,73 @@ jobs:
...
+          # Duplicate the main config.
+          yq ".promotion.tag=\"$RELEASE\" | .zz_generated_metadata.branch=\"release-$RELEASE\"" \
+            "$CFG_DIR/stackrox-stackrox-master.yaml" \
+            > "${PREFIX}.yaml"
+
+          # Duplicate the opensource images config.
+          yq ".promotion.tag=\"$RELEASE-stackrox-branding\" | .zz_generated_metadata.branch=\"release-$RELEASE\"" \
+            "$CFG_DIR/stackrox-stackrox-master__stackrox_branding.yaml" \
+            > "${PREFIX}__stackrox_branding.yaml"
+
```
Comment (gavin-stackrox) [inline .github/workflows/start-release.yml:155]: I think you are going to need to `yq` out all the `tests` that have no `postsubmit: true` line and `cron: x` or openshift CI will fail. But I may be wrong and maybe it is ok with deleting the job files after the `make update`. Have you been able to test with a `0.x` release?

**Review by vikin91 — COMMENTED** [review]
This looks okay at the first glimpse, but I would like to see how you tested it and what was the result - I think there is no harm in creating that branch earlier for some testing release?

**Review by vikin91 — APPROVED** [review]
We need to start testing things in some wetter runs :) When are we introducing the weekly upstream releases? 😬

File: .github/workflows/start-release.yml (hunk 1 above)
Comment (0x656b694d) [reply .github/workflows/start-release.yml:155]: Oh, I didn't test it end-to-end, I just refactored the script you documented and observed `make update` doesn't fail. I'll need your assistance in broader testing.

Comment (0x656b694d) [reply .github/workflows/start-release.yml:155]: The `presubmits` files are still getting generated after cutting out all the 'non-postsumbit' and 'cron' tests with `--target=[images]`. So I'll keep the `rm` step.

Comment (0x656b694d) [reply .github/workflows/start-release.yml:159]: https://github.com/openshift/release/pull/31722

**Comment by gavin-stackrox** [comment]
I'd like to see this run on a test release branch to see what happens. I've used a `0.` prefix as a test release indicator which e.g. sends slack messages to different channels: https://github.com/stackrox/stackrox/blob/4b3c61e55330bade1ad326ab6cd597623da0a29a/scripts/lib.sh#L51

**Pull request merged**